source/extensions/filters/network/thrift_proxy/metadata.h

```diff
@@ -163,6 +167,9 @@ class MessageMetadata {
   bool isDraining() const { return is_draining_; }
   void setDraining(bool draining) { is_draining_ = draining; }
 
+  bool fromClient() const { return from_client_; }
+  void setFromClient(bool from_client) { from_client_ = from_client; }
```

Comment:
Likewise `isRequest`. Also thinking that it doesn't seem that a Metadata instance switches from request to response? Maybe we can just set this at construction time and have it be constant.

Reply:
I think we probably can, yeah.